submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/SubmissionOperations.kt

```diff
@@ -66,21 +68,21 @@ interface SubmissionRequestPersistenceService {
 
     fun getPendingRequest(accNo: String, version: Int): SubmissionRequest
 
+    fun getIndexedRequest(accNo: String, version: Int): SubmissionRequest
+
     fun getLoadedRequest(accNo: String, version: Int): SubmissionRequest
 
     fun getCleanedRequest(accNo: String, version: Int): SubmissionRequest
 
     fun getRequestStatus(accNo: String, version: Int): RequestStatus
 }
 
-interface SubmissionFilesPersistenceService {
-    fun saveSubmissionFile(file: SubmissionFile)
-
-    fun getSubmissionFile(path: String, accNo: String, version: Int): ExtFile
+interface SubmissionRequestFilesPersistenceService {
+    fun upsertSubmissionRequestFile(file: SubmissionRequestFile)
 
-    fun getSubmissionFiles(accNo: String, version: Int, startingAt: Int): List<Pair<ExtFile, Int>>
+    fun getSubmissionRequestFile(accNo: String, version: Int, filePath: String): SubmissionRequestFile
 
-    fun getFileListFiles(accNo: String, version: Int, fileListName: String): List<ExtFile>
+    fun getSubmissionRequestFiles(accNo: String, version: Int, startingAt: Int): Stream<SubmissionRequestFile>
```

Comment:
I would suggest to use kotlin extension `asSequence` so you get all the benefitcs of kotlin rather than use java data structure